### DISCUSSION TITLE
Set home position

### DISCUSSION THREAD

**Discussion by vhristenco** [Jyers/Marlin#835] [Q&A] [answered]
I recently compiled the latest firmware and found this option under Homing. 
What's it supposed to do? Moving to a position and then enabling "Set Home Position" doesn't change anything.

**Comment by gothcha** [comment, marked as answer]
I believe this was added for laser use where the Z height needs to start at an arbitrary point. It isn’t something you need to be concerned about for normal printing.

**Reply by vhristenco** [reply]
Yeah, by looking at the .cpp I see it sends a G92.
However, it might be useful if you change it to compute the x and y nozzle offset and send it with M206.

**Reply by Jyers** [reply]
This would work if marlin didn't limit the xy offsets to 20mm max... So unfortunately the only way to do this is to use G92 to set the current position to 0 0

**Reply by IrishBrewer** [reply]
> This would work if marlin didn't limit the xy offsets to 20mm max... So unfortunately the only way to do this is to use G92 to set the current position to 0 0

With V1.3.4, my Z home position seems to be zero and I would like it to be off the bed so it doesn't leave blobs and/or clog the nozzle.  If I move Z to 10 and then Set Home Position, it seems like it should reset the Z home position.  Or is there another way to set it with the precompiled firmware?

**Reply by Jyers** [reply]
I don't think the home position does what you think. Setting the home position or home offsets doesn't change where the printhead is when it hits an endstop, it just changes where it thinks it is. If you want to keep your nozzle off the bed during a print warmup or such i'd recommend adding a z lift to your start gcode somewhere after homing.

**Reply by IrishBrewer** [reply]
Is there a way in the UI to move the saved home position up when you home all or home Z?  I can add some gcode to my prints but I would like to have it moved for UI homing also.  I usually do my preheating from the UI and start the print separately.

**Reply by gothcha** [reply]
@Jyers There has been lots of confusion on the Set Home Position function, would it make sense to change the title to something like ‘Set Origin to Nozzle’ then in the status line display ‘Print origin set to current nozzle XYZ’
